w3af/plugins/grep/csp.py

```diff
@@ -19,158 +19,163 @@
+        if csp_vulns:
+            result += self.vuln_csp_tpl + csp_vulns
+        if self._responses_without_csp:
+            result += self.vuln_without_tpl % "\n".join(
+                    [r[1] for r in self._responses_without_csp])
+        result += self.vuln_footer_tpl
+
+        v = Vuln(self.VULN_NAME, result, severity.MEDIUM, 
+                [csp.response_id for csp in self.csps],
+                self.get_name())
+        self.kb_append(self, 'csp', v)
+
+    def get_options(self):
         """
-        list_resp_id = []
-
-        # Check input for quick exit
-        if vuln_desc is None or vuln_desc.strip() == "":
-            return list_resp_id
-       
-        # Parse vulns collection
-        ref = vuln_desc.lower().strip()        
-        for vuln_store_item in self._vulns:
-            for csp_directive_name, csp_vulns_list in vuln_store_item.csp_vulns.iteritems():
-                for csp_vuln in csp_vulns_list:        
-                    if csp_vuln.desc.strip().lower() == ref:
-                        if vuln_store_item.resp_id not in list_resp_id:
-                            list_resp_id.append(vuln_store_item.resp_id)
+        :return: A list of option objects for this plugin.
+        """
+        ol = OptionList()
 
-        return list_resp_id
+        d = 'Maximum number of sources in source list'
+        o = opt_factory('source_limit', self._source_limit, d, 'integer')
+        ol.add(o)
 
+        d = 'Trusted hosts for finding untrusted ones in source list'
+        o = opt_factory('trusted_hosts', self._trusted_hosts, d, 'list')
+        ol.add(o)
 
-class DiskCSPVulnStoreItem(DiskItem):
-    """
-    This is a class to store CSP vulnerabilities found for a URL (URL+ID) in a
-    DiskList or DiskSet.
-    """
+        return ol
 
-    def __init__(self, r_url, r_id, r_vulns):
+    def set_options(self, option_list):
         """
-        Constructor.
-        @param r_url: HTTP response url
-        @param r_id: HTTP response ID
-        @param r_vulns: CSP vulnerabilities found
+        This method sets all the options that are configured using the user
+        interface generated by the framework using the result of get_options().
+        
+        :param options_list: A dictionary with the options for the plugin.
+        :return: No value is returned.
         """
-        self.url = r_url
-        self.resp_id = r_id
-        self.csp_vulns = r_vulns
 
-    def get_eq_attrs(self):
+        self._source_limit = option_list['source_limit'].get_value()
+        self._trusted_hosts = option_list['trusted_hosts'].get_value()
+
+    def get_long_desc(self):
         """
-        Implements method from base class.
+        :return: A DETAILED description of the plugin functions and features.
         """
-        return ['url','resp_id']
\ No newline at end of file
+        return """
+        This plugin identifies incorrect or too permissive Content Security Policy version 2 
```

Comment:
Is version 2 back compatible with CSP version 1? Maybe the description should say "too permissive Content Security Policy versions 1 and 2" ?